### ISSUE TITLE
Claude 3 ReAct Support

### ISSUE THREAD

**Issue by oberlinstands** [run-llama/LlamaIndexTS#660]
I see in the docs Claude 3 Opus is not officially supported in the ReAct agent. Curious to see if this is on the roadmap, if someone is working on it, or what I could do to get started by opening up a PR?

**Comment by marcusschiesser** [comment]
Thanks, @oberlinstands. That's a great idea and we're always happy for any PR!
Using the ReAct Agent should be possible with any LLM, including Claude 3. 

I just created a PR with an example https://github.com/run-llama/LlamaIndexTS/pull/661 - you might want to start from that.

Running it shows up the following error:
```
Thought: I need to use the divideNumbers tool to divide 16 by 2, then use the sumNumbers tool to add 20 to the result.
Action: divideNumbers
Action Input: {"a":16,"b":2}

Observation: 8
/Users/marcus/code/llamaindex/LlamaIndexTS/node_modules/.pnpm/@anthropic-ai+sdk@0.18.0/node_modules/@anthropic-ai/sdk/src/error.ts:58
      return new BadRequestError(status, error, message, headers);
             ^
BadRequestError: 400 {"type":"error","error":{"type":"invalid_request_error","message":"messages: roles must alternate between \"user\" and \"assistant\", but found multiple \"user\" roles in a row"}}
    at Function.generate (/Users/marcus/code/llamaindex/LlamaIndexTS/node_modules/.pnpm/@anthropic-ai+sdk@0.18.0/node_modules/@anthropic-ai/sdk/src/error.ts:58:14)
    at Anthropic.makeStatusError (/Users/marcus/code/llamaindex/LlamaIndexTS/node_modules/.pnpm/@anthropic-ai+sdk@0.18.0/node_modules/@anthropic-ai/sdk/src/core.ts:383:21)
    at Anthropic.makeRequest (/Users/marcus/code/llamaindex/LlamaIndexTS/node_modules/.pnpm/@anthropic-ai+sdk@0.18.0/node_modules/@anthropic-ai/sdk/src/core.ts:446:24)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Anthropic.chat (/Users/marcus/code/llamaindex/LlamaIndexTS/packages/core/dist/cjs/llm/LLM.js:596:26)
    at async ReActAgentWorker._runStep (/Users/marcus/code/llamaindex/LlamaIndexTS/packages/core/dist/cjs/agent/react/worker.js:227:30)
    at async ReActAgentWorker.runStep (/Users/marcus/code/llamaindex/LlamaIndexTS/packages/core/dist/cjs/agent/react/worker.js:248:16)
    at async ReActAgent._runStep (/Users/marcus/code/llamaindex/LlamaIndexTS/packages/core/dist/cjs/agent/runner/base.js:124:29)
    at async ReActAgent._chat (/Users/marcus/code/llamaindex/LlamaIndexTS/packages/core/dist/cjs/agent/runner/base.js:186:35)
    at async ReActAgent.chat (/Users/marcus/code/llamaindex/LlamaIndexTS/packages/core/dist/cjs/agent/runner/base.js:201:30)
    at async main (/Users/marcus/code/llamaindex/LlamaIndexTS/examples/agent/react_agent.ts:72:20) {
  status: 400,
  headers: {
    'cf-cache-status': 'DYNAMIC',
    'cf-ray': '8682fe967f0b4b77-BKK',
    connection: 'keep-alive',
    'content-length': '179',
    'content-type': 'application/json',
    date: 'Fri, 22 Mar 2024 03:11:28 GMT',
    'request-id': 'req_0162yTuRmEbwQn6kL3G4pieW',
    server: 'cloudflare',
    via: '1.1 google',
    'x-cloud-trace-context': '3d199eef995d8a09c53340a105161ead',
    'x-should-retry': 'false'
  },
  error: {
    type: 'error',
    error: {
      type: 'invalid_request_error',
      message: 'messages: roles must alternate between "user" and "assistant", but found multiple "user" roles in a row'
    }
  }
}
```

I would start by looking at `ReActAgentWorker.runStep` - it seems like it's sending a wrong `messages` array to the LLM.

If you need any help, @EmanuelCampos created the React Agent.

**Comment by EmanuelCampos** [comment]
@oberlinstands the fix was done on #661 + an example, we should be releasing it soon, feel free to test on main for now

**Comment by oberlinstands** [comment]
Wow, you guys are fast. Will do, ty!

**Comment by EmanuelCampos** [comment]
Fixed on llamaindex@0.2.2
